runtime/src/tasks/qthreads/tasks-qthreads.c

```diff
@@ -523,20 +523,24 @@ static void setupAvailableParallelism(int32_t maxThreads) {
 }
 
 static chpl_bool setupGuardPages(void) {
+    const char *armArch = "arm-thunderx";
     chpl_bool guardPagesEnabled = true;
     // default value set by compiler (--[no-]stack-checks)
     chpl_bool defaultVal = (CHPL_STACK_CHECKS == 1);
 
     // Setup guard pages. Default to enabling guard pages, only disabling them
     // under the following conditions (Precedence high-to-low):
-    // 1) Guard pages disabled at configure time
-    // 2) Guard pages not supported because of huge pages
-    // 3) QT_GUARD_PAGES set to a 'false' value
-    // 4) CHPL_STACK_CHECKS set (--no-stack-checks thrown at compilation time)
+    //  - Guard pages disabled at configure time
+    //  - Guard pages not supported because of huge pages
+    //  - Guard pages not supported by processor
+    //  - QT_GUARD_PAGES set to a 'false' value
+    //  - CHPL_STACK_CHECKS set (--no-stack-checks thrown at compilation time)
     if (!CHPL_QTHREAD_HAVE_GUARD_PAGES) {
         guardPagesEnabled = false;
     } else if (chpl_getHeapPageSize() != chpl_getSysPageSize()) {
         guardPagesEnabled = false;
+    } else if (strncmp(armArch, CHPL_TARGET_ARCH, strlen(armArch)) == 0) {
```

Comment:
I'm also happy to switch this to `strstr("arm-thunderx", CHPL_TARGET_ARCH) != NULL` if you find that to be a little cleaner.

Reply:
Thanks.  I'm not going to ask for a change.

The way would normally do it is to make armArch an array instead of a pointer, and then I can say sizeof(armArch) - 1 and not have to do a strlen.